doc/API.md

```diff
@@ -146,3 +157,212 @@ Failing requests may be due to the following errors:
 * status code 400, errno 103: Missing or malformed authentication header.
 * status code 400, errno 400: Bad request.
 * status code 401, errno 401: Unauthorized. If credentials are not valid.
+
+## POST /users
+Create a new inactive user registration and sends an activation email to the user specified email.
+
+Only users with admin privileges are able to access this method.
+
+### Request
+Requests must include an authorization header containing a [bearer token](#authentication).
+
+___Parameters___
+* email - User email.
+
+```ssh
+POST /users HTTP/1.1
+Content-Type: application/json
+Authorization: Bearer QWxhZGRpbjpPcGVuU2VzYW1l...
+{
+  "email": "user@domain.org"
+}
+```
+### Response
+Successful requests will produce a "201 Created" response with a body containing an activation url:
+```ssh
+HTTP/1.1 201 Created
+Connection: close
+{
+  "activation_url": "/v1/users/InR5cCI6IkpXVCJ"
+}
+```
+
+Failing requests may be due to the following errors:
+* status code 400, errno 101: Invalid email. Missing or malformed email.
+* status code 400, errno 400: Bad request.
+* status code 409, errno 409: Already exists.
+* status code 401, errno 401: Unauthorized. If credentials are not valid.
+
+## GET /users
+Get the list of all registered users.
+
+Only users with admin privileges are able to access this method.
+
+### Request
+Requests must include an authorization header containing a [bearer token](#authentication).
+```ssh
+GET /users HTTP/1.1
+Authorization: Bearer QWxhZGRpbjpPcGVuU2VzYW1l...
+```
+
+### Response
+Successful requests will produce a "200 OK" response with a body containing the list of all registered users:
+```ssh
+HTTP/1.1 200 OK
+Connection: close
+[{
+  "id": "hfkjsIklksadhs",
+  "username": "admin"
+  "email": "admin@domain.org",
+  "is_admin": true,
+  "is_active": true
+}, {
+  "id": "InR5cCI6IkpXVCJ",
+  "username": "pepe"
+  "email": "user@domain.org",
+  "is_admin": false,
+  "is_active": true
+}, {
+  "id": "kjfIsalj5893213",
+  "email": "another_user@domain.org",
+  "is_admin": false,
+  "is_active": false
+}]
+```
+
+Failing requests may be due to the following errors:
+* status code 400, errno 400: Bad request.
+* status code 401, errno 401: Unauthorized. If credentials are not valid.
+
+## GET /users/:id
+Get the information of the user matching the given id.
```

Comment:
The use case for this is to allow the user to modify its own information. Before editing the information, the user needs to see it :) Once we have permissions, we will forbid a user to get information about another user unless the one requesting the information has admin permissions.